compiler/ballerina-lang/src/main/resources/compiler.properties

```diff
@@ -371,6 +371,12 @@ error.missing.required.record.field=\
 error.default.values.not.allowed.for.optional.fields=\
   a default value specified for optional field ''{0}''
 
+error.never.type.not.allowed.for.required.fields=\
+  field ''{0}'' is defined as a never-typed required field
```

Comment:
Better to define error messages so that it clearly conveys that what's being attempted is not allowed, and if possible, what to use instead.

Reply:
How about this ?
`'never' type is not allowed as a required field, use optional field declaration for ''{0}'' instead`